hazelcast/src/main/java/com/hazelcast/config/CollectionConfig.java

```diff
@@ -58,8 +58,8 @@
     private int asyncBackupCount = DEFAULT_ASYNC_BACKUP_COUNT;
     private int maxSize = DEFAULT_MAX_SIZE;
     private boolean statisticsEnabled = true;
-
     private String quorumName;
+    private MergePolicyConfig mergePolicyConfig = new MergePolicyConfig();
```

Comment:
Is the new `mergePolicy` config property covered by the client protocol for dynamically adding config? Is this in a different PR?
(see: `AddListConfigMessageTask`, `AddSetConfigMessageTask`, `ClientDynamicClusterConfig#addSetConfig`, `ClientDynamicClusterConfig#addListConfig`)

Reply:
I have no idea how this works (both, never did anything with the Client Protocol and that part of the dynamic config is also new to me). But we have a similar open issue for other configs (https://github.com/hazelcast/hazelcast/issues/12126) and Thomas mentioned in his PR, that we anyway need to merge this first, before the Client Protocol can be modified. I would say we'll merge all server-side implementations and then add the merge policies to the Client Protocol in a single PR. WDYT?

Reply:
Basically we need to allow adding new configuration from the client when the cluster is running. To allow specifying the merge policy, the client protocol needs to be expanded, e.g. https://github.com/mmedenjak/hazelcast-client-protocol/blob/master/hazelcast/src/main/java/com/hazelcast/client/impl/protocol/template/DynamicConfigTemplate.java#L118-L135

Reply:
Merging this PR and addressing it later in a separate PR is fine by me, BTW.